dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/redshift/RedshiftClusterUsageMetricsTask.java

```diff
@@ -0,0 +1,213 @@
+                    getClusterMetrics(cluster.getClusterIdentifier()).entrySet().stream()
+                        .map(
+                            metricEntry ->
+                                Stream.of(
+                                        new Object[] {
+                                          cluster.getClusterIdentifier(), metricEntry.getKey()
+                                        },
+                                        metricEntry.getValue())
+                                    .flatMap(Stream::of)
+                                    .toArray()))
+            .collect(toList()));
+    return null;
+  }
+
+  private ImmutableList<Cluster> listClusters() {
+    AmazonRedshift client = redshiftApiClient();
+    return ImmutableList.copyOf(
+        client.describeClusters(new DescribeClustersRequest()).getClusters());
+  }
+
+  private ImmutableList<MetricDataPoint> getMetricDataPoints(
+      String clusterId, MetricConfig metricConfig) {
+    AmazonCloudWatch client = cloudWatchApiClient();
+    GetMetricStatisticsRequest request =
+        new GetMetricStatisticsRequest()
+            .withMetricName(metricConfig.name().name())
+            .withStatistics(metricConfig.type().name())
+            .withNamespace(REDSHIFT_NAMESPACE)
+            .withDimensions(new Dimension().withName("ClusterIdentifier").withValue(clusterId))
+            .withStartTime(Date.from(interval.getStartUTC().toInstant()))
+            .withEndTime(Date.from(interval.getEndExclusiveUTC().toInstant()))
+            .withPeriod((int) metricDataPeriod().getSeconds());
+
+    GetMetricStatisticsResult result = client.getMetricStatistics(request);
+    return result.getDatapoints().stream()
+        .map(
+            datapoint ->
+                MetricDataPoint.create(
+                    datapoint.getTimestamp(), getDatapointValue(metricConfig, datapoint)))
+        .collect(toImmutableList());
+  }
+
+  private Double getDatapointValue(MetricConfig metricConfig, Datapoint datapoint) {
+    switch (metricConfig.type()) {
+      case Average:
+        return datapoint.getAverage();
+      default:
+        return null;
+    }
+  }
+
+  private ImmutableSortedMap<String, Double[]> getClusterMetrics(String clusterIdentifier) {
+    return zipMetrics(
+        metrics.stream()
+            .map(metricConfig -> getMetricDataPoints(clusterIdentifier, metricConfig))
+            .collect(toImmutableList()));
+  }
+
+  private ImmutableSortedMap<String, Double[]> zipMetrics(
+      ImmutableList<ImmutableList<MetricDataPoint>> metricsCollection) {
+    SortedMap<String, Double[]> metricsMap = new TreeMap<String, Double[]>();
+
+    metricsCollection.stream()
+        .forEach(
+            metricDatapoints -> {
+              int metricIndex = metricsCollection.indexOf(metricDatapoints);
+
+              metricDatapoints.forEach(
+                  dataPoint -> {
+                    String dateString = DATE_FORMAT.format(dataPoint.date().toInstant());
```

Comment:
I would suggest changing the type of key in the map to `Instant` and moving `DATE_FORMAT.format` closer to the serialization to CSV row.